go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/dapr/kit
 
-go 1.20
+go 1.22
```

Comment:
There's no need to update the Go version in dapr/kit for the sake of updating it, unless we need to depend on features added in 1.21+?